extensions/kafka-streams/runtime/src/main/java/io/quarkus/kafka/streams/runtime/KafkaStreamsTopologyManager.java

```diff
@@ -80,17 +84,90 @@ private static Properties getStreamsProperties(Properties properties, String boo
+        if (runtimeConfig.schemaRegistryUrl.isPresent()) {
+            streamsProperties.put(runtimeConfig.schemaRegistryKey, runtimeConfig.schemaRegistryUrl.get());
+        }
+
+        // sasl
+        SaslConfig sc = runtimeConfig.sasl;
+        if (sc != null) {
+            setProperty(sc.jaasConfig, streamsProperties, SaslConfigs.SASL_JAAS_CONFIG);
+
+            setProperty(sc.clientCallbackHandlerClass, streamsProperties, SaslConfigs.SASL_CLIENT_CALLBACK_HANDLER_CLASS);
+
+            setProperty(sc.loginCallbackHandlerClass, streamsProperties, SaslConfigs.SASL_LOGIN_CALLBACK_HANDLER_CLASS);
+            setProperty(sc.loginClass, streamsProperties, SaslConfigs.SASL_LOGIN_CLASS);
+
+            setProperty(sc.kerberosServiceName, streamsProperties, SaslConfigs.SASL_KERBEROS_SERVICE_NAME);
+            setProperty(sc.kerberosKinitCmd, streamsProperties, SaslConfigs.SASL_KERBEROS_KINIT_CMD);
+            setProperty(sc.kerberosTicketRenewWindowFactor, streamsProperties,
+                    SaslConfigs.SASL_KERBEROS_TICKET_RENEW_WINDOW_FACTOR);
+            setProperty(sc.kerberosTicketRenewJitter, streamsProperties, SaslConfigs.SASL_KERBEROS_TICKET_RENEW_JITTER);
+            setProperty(sc.kerberosMinTimeBeforeRelogin, streamsProperties, SaslConfigs.SASL_KERBEROS_MIN_TIME_BEFORE_RELOGIN);
+
+            setProperty(sc.loginRefreshWindowFactor, streamsProperties, SaslConfigs.SASL_LOGIN_REFRESH_WINDOW_FACTOR);
+            setProperty(sc.loginRefreshWindowJitter, streamsProperties, SaslConfigs.SASL_LOGIN_REFRESH_WINDOW_JITTER);
+
+            setProperty(sc.loginRefreshMinPeriod, streamsProperties, SaslConfigs.SASL_LOGIN_REFRESH_MIN_PERIOD_SECONDS,
+                    DurationToSecondsFunction.INSTANCE);
+            setProperty(sc.loginRefreshBuffer, streamsProperties, SaslConfigs.SASL_LOGIN_REFRESH_BUFFER_SECONDS,
+                    DurationToSecondsFunction.INSTANCE);
+        }
+
+        // ssl
+        SslConfig ssl = runtimeConfig.ssl;
+        if (ssl != null) {
+            setProperty(ssl.protocol, streamsProperties, SslConfigs.SSL_PROTOCOL_CONFIG);
+            setProperty(ssl.provider, streamsProperties, SslConfigs.SSL_PROVIDER_CONFIG);
+            setProperty(ssl.cipherSuites, streamsProperties, SslConfigs.SSL_CIPHER_SUITES_CONFIG);
+            setProperty(ssl.enabledProtocols, streamsProperties, SslConfigs.SSL_ENABLED_PROTOCOLS_CONFIG);
+
+            setStoreConfig(ssl.truststore, streamsProperties, "ssl.truststore");
+            setStoreConfig(ssl.keystore, streamsProperties, "ssl.keystore");
+            setStoreConfig(ssl.key, streamsProperties, "ssl.key");
+
+            setProperty(ssl.keymanagerAlgorithm, streamsProperties, SslConfigs.SSL_KEYMANAGER_ALGORITHM_CONFIG);
+            setProperty(ssl.trustmanagerAlgorithm, streamsProperties, SslConfigs.SSL_TRUSTMANAGER_ALGORITHM_CONFIG);
+            setProperty(ssl.endpointIdentificationAlgorithm.map(s -> "\"\"".equals(s) ? "" : s), streamsProperties,
```

Comment:
```suggestion
            setProperty(ssl.endpointIdentificationAlgorithm.orElse(""), streamsProperties,
```

Reply:
How are you gonna get Optional.empty() if default is "https" ... ?
Or it looks like I don't understand how @ConfigItem with defaultValue and Optional work together?

Reply:
An empty `Optional` means the value was empty - meaning, the user explicitly gave an empty value.  If the user *does not give* a value, then the default is used, so in this case you'd get an `Optional.of("https")`.